tests/test_export_onnx.py

```diff
@@ -0,0 +1,183 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+
+import io
+import unittest
+import torch
+from torch.hub import _check_module_exists
+
+from detectron2 import model_zoo
+from detectron2.config import get_cfg
+from detectron2.export import STABLE_ONNX_OPSET_VERSION
+from detectron2.export.flatten import TracingAdapter
+from detectron2.modeling import build_model
+from detectron2.utils.testing import (
+    _pytorch1111_symbolic_opset9_repeat_interleave,
+    _pytorch1111_symbolic_opset9_to,
+    get_sample_coco_image,
+    min_torch_version,
+    register_custom_op_onnx_export,
+    skipIfOnCPUCI,
+    skipIfUnsupportedMinOpsetVersion,
+    skipIfUnsupportedMinTorchVersion,
+    unregister_custom_op_onnx_export,
+)
+
+
+@unittest.skipIf(not _check_module_exists("onnx"), "ONNX not installed.")
+@unittest.skipIf(
+    not min_torch_version("1.10"),
+    f"module 'torch' has __version__ " f"{torch.__version__}, required is: 1.10",
+)
+class TestONNXTracingExport(unittest.TestCase):
+    def testMaskRCNNFPN(self):
+        def inference_func(model, images):
+            inputs = [{"image": image} for image in images]
+            inst = model.inference(inputs, do_postprocess=False)[0]
+            return [{"instances": inst}]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml", inference_func
+        )
+
+    @skipIfOnCPUCI
+    def testMaskRCNNC4(self):
+        def inference_func(model, image):
+            inputs = [{"image": image}]
+            return model.inference(inputs, do_postprocess=False)[0]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_C4_3x.yaml", inference_func
+        )
+
+    @skipIfOnCPUCI
+    def testCascadeRCNN(self):
+        def inference_func(model, image):
+            inputs = [{"image": image}]
+            return model.inference(inputs, do_postprocess=False)[0]
+
+        self._test_model_zoo_from_config_path(
+            "Misc/cascade_mask_rcnn_R_50_FPN_3x.yaml", inference_func
+        )
+
+    def testRetinaNet(self):
+        def inference_func(model, image):
+            return model.forward([{"image": image}])[0]["instances"]
+
+        self._test_model_zoo_from_config_path(
+            "COCO-Detection/retinanet_R_50_FPN_3x.yaml", inference_func
+        )
+
+    @skipIfOnCPUCI
+    def testMaskRCNNFPN_batched(self):
+        def inference_func(model, image1, image2):
+            inputs = [{"image": image1}, {"image": image2}]
+            return model.inference(inputs, do_postprocess=False)
+
+        self._test_model_zoo_from_config_path(
+            "COCO-InstanceSegmentation/mask_rcnn_R_50_FPN_3x.yaml", inference_func, batch=2
+        )
+
+    @skipIfUnsupportedMinOpsetVersion(16, STABLE_ONNX_OPSET_VERSION)
```

Comment:
Please add a todo here to update the value of `STABLE_ONNX_OPSET_VERSION` so this check can be removed in the future.

`skipIfUnsupportedMinOpsetVersion` is used only once, so it's easier to just write `@unittest.skipIf( STABLE_ONNX_OPSET_VERSION < 16, "torch version too old")`

Reply:
It is used once because ONNX export functionality is divided into several PRs.

https://github.com/facebookresearch/detectron2/pull/4205 uses `skipIfUnsupportedMinOpsetVersion` at least 2 times and possibly more when I am done